lua/treesitter-context.lua

```diff
@@ -67,8 +67,10 @@ local function cannot_open(bufnr, winid)
     or api.nvim_win_get_height(winid) < config.min_window_height
 end
 
+-- Always use deferred execution to prevent from segfaults in neovim, like this one:
+-- https://github.com/neovim/neovim/issues/31236
 ---@param winid integer
-local update_single_context = throttle_by_id(function(winid)
+local update_single_context = throttle_by_id(vim.schedule_wrap(function(winid)
```

Comment:
I think this doesn't work with throttle since vim.schedule is non-blocking.

Reply:
Is it acceptable for the throttler to allow the following scenario?
```
  Time ->
|cb|-----timeout-------|cb||cb|-----timeout-----|
```

Reply:
I fixed this bug. It is now guaranteed that between two `|cb|` executing there is at least 150 ms gap.